File: game_plugin_interface/src/sc/protocol/RoomPacket.java
package sc.protocol;

import sc.protocol.requests.ILobbyRequest;

public class RoomPacket implements ILobbyRequest
{
	private String	roomId;
	private Object	data;

	public RoomPacket(String roomId, Object o)
	{
		this.roomId = roomId;
		this.data = o;
	}

	public String getRoomId()
	{
		return this.roomId;
	}

	public Object getData()
	{
		return this.data;
	}
}

File: sdk/src/server-api/sc/protocol/room/RoomPacket.kt
package sc.protocol.room

import com.thoughtworks.xstream.annotations.XStreamAlias
import com.thoughtworks.xstream.annotations.XStreamAsAttribute
import sc.protocol.ProtocolPacket

/** Used to send a [RoomMessage] to a room. */
@XStreamAlias("room")
data class RoomPacket(
        @XStreamAsAttribute
        val roomId: String,
        val data: RoomMessage,
): ProtocolPacket
